### PR TITLE
fix: Stuck unbondings

### PR DESCRIPTION
**Pull request by kruspy**

### CODE REVIEW

**Review by xlab — APPROVED** [review]

File: x/liquidstakeibc/keeper/hooks.go [hunk 1]
```diff
@@ -637,7 +654,12 @@ func (k *Keeper) UndelegationWorkflow(ctx sdk.Context, epoch int64) {
 				"host_chain",
 				hc.ChainId,
 			)
-			return
+
+			// mark the unbonding as failed
+			unbonding.State = liquidstakeibctypes.Unbonding_UNBONDING_FAILED
+			k.SetUnbonding(ctx, unbonding)
```
Comment (puneet2019) [inline x/liquidstakeibc/keeper/hooks.go:660]: add an unbonding state event? so we can track it?

**Comment by kruspy** [comment]
I am moving the treating of old events to a migration.

File: x/liquidstakeibc/migrations/v4/store.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package v4
```
Comment (puneet2019) [inline x/liquidstakeibc/migrations/v4/store.go:1]: uh, can we not just write these parts as upgrade instructions? and not increase module consensus version? 
Module consensus version is usually changed when store structs change. not when we edit/ fork entries

File: app/app.go [hunk 3]
```diff
@@ -1021,19 +1021,4 @@ func (app *PstakeApp) RegisterUpgradeHandler() {
...
 	)
-
-	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
-	if err != nil {
-		panic(err)
-	}
-
-	if upgradeInfo.Name == UpgradeName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
-		storeUpgrades := store.StoreUpgrades{
-			Added:   []string{ratesynctypes.StoreKey, liquidstaketypes.StoreKey},
```
Comment (puneet2019) [inline app/app.go:1032]: keep this blob in general, just remove the `Added: ` things.. 
anyway everything is under if statements, it won't trigger unless there is an upgrade.

**Review by puneet2019 — APPROVED** [review]
lgtm, test, and add test-cases for regression before we merge

**Pull request merged**